x-pack/plugins/apm/public/components/app/Main/UpdateBreadcrumbs.ts

```diff
@@ -26,11 +31,16 @@ interface Props {
 class UpdateBreadcrumbsComponent extends React.Component<Props> {
   public updateHeaderBreadcrumbs() {
     const { _g = '', kuery = '' } = toQuery(this.props.location.search);
-    const breadcrumbs = this.props.breadcrumbs.map(({ breadcrumb, match }) => ({
-      text: breadcrumb,
-      href: `#${match.url}?_g=${_g}&kuery=${kuery}`
-    }));
+    const breadcrumbs = this.props.breadcrumbs.map(({ breadcrumb, match }) => {
+      const text =
+        typeof breadcrumb === 'string'
+          ? breadcrumb
+          : breadcrumb.type(breadcrumb.props); // "render" the element
```

Comment:
Wow, this made me look twice. But thinking about it, I really like it. Is it something we should expect to break when updating React, or is this approach supported?
If we can get official types that would be awesome!

Reply:
Btw. this is a really useful blog post in the regard: https://reactjs.org/blog/2015/12/18/react-components-elements-and-instances.html